From 4b3c5ffcc8b1ceb2c25e8f1c1e10700b5e582aa0 Mon Sep 17 00:00:00 2001
From: Jan Remmet <j.remmet@phytec.de>
Date: Thu, 19 Nov 2015 13:58:51 +0100
Subject: [PATCH] net: eth: Remove ethaddr_param
Organization: Develer S.r.l.

commit d00db554 introduce ethaddr_param. But it is not initialized, so
devinfo fails to show the mac. Remove it and use edev->ethaddr.

Signed-off-by: Jan Remmet <j.remmet@phytec.de>
Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
(cherry picked from commit b9170a1bde192c691f15c4474ed2f1e7bfebbb41)
---
 include/net.h | 1 -
 net/eth.c     | 4 ++--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/include/net.h b/include/net.h
index d7a4751..2a37a43 100644
--- a/include/net.h
+++ b/include/net.h
@@ -60,7 +60,6 @@ struct eth_device {
 	IPaddr_t serverip;
 	IPaddr_t netmask;
 	IPaddr_t gateway;
-	char ethaddr_param[6];
 	char ethaddr[6];
 };
 
diff --git a/net/eth.c b/net/eth.c
index b22e556..6ab9908 100644
--- a/net/eth.c
+++ b/net/eth.c
@@ -285,7 +285,7 @@ static int eth_param_set_ethaddr(struct param_d *param, void *priv)
 {
 	struct eth_device *edev = priv;
 
-	return eth_set_ethaddr(edev, edev->ethaddr_param);
+	return eth_set_ethaddr(edev, edev->ethaddr);
 }
 
 #ifdef CONFIG_OFTREE
@@ -369,7 +369,7 @@ int eth_register(struct eth_device *edev)
 	dev_add_param_ip(dev, "gateway", NULL, NULL, &edev->gateway, edev);
 	dev_add_param_ip(dev, "netmask", NULL, NULL, &edev->netmask, edev);
 	dev_add_param_mac(dev, "ethaddr", eth_param_set_ethaddr, NULL,
-			edev->ethaddr_param, edev);
+			edev->ethaddr, edev);
 
 	if (edev->init)
 		edev->init(edev);
-- 
1.9.1

